docs/decisions/08-abac-claims.md

```diff
@@ -0,0 +1,98 @@
+# Authorization Claims
+
+Letterbook makes use of a attribute based authorization scheme. Attributes are primarily (but likely not exclusively) expressed as claims. This describes that scheme, and an initial set of claims to work from.
+
+## Status
+
+- [ ] Decided
+- [ ] Decided Against
+- [ ] Deferred
+- [ ] Superseded
+
+# Decision
+
+A summary of what was proposed and decided
+
+## Claims
+
+The initial set of claims to work from
+
+Profile 
+- `Rename` (actions related to the display name and handle)
+- `Edit` (actions related to other fields like description and pfp)
+- `Follow` (follow other profiles)
+- `Unfollow` (stop following other propfiles)
+- `ApproveFollower` (approve, reject, or remove followers)
+- `Block` (actions related to moderation tools like block, mute, etc)
+- `ReadNotif` (view the notifications sent to the profile)
+- `ClearNotif` (mark notifications as read)
+- `Export` (actions related to exporting bulk data about the profile)
+- `Sign` (signing client-provided data with a server-managed key)
+- `ManageKeys` (add or revoke signing keys)
+- `Grant` (grant claims to other agents)
+- `Guest` (other claims are delegated, and some highly sensitive actions should still be prohibited)
+- `Owner` (holder is the owner, useful to help limit the size of tokens)
+- Post
+  - `Draft`
+  - `Publish`
+  - `Delete`
+  - `Read`
+  - `ReadDraft`
+  - `React`
+  - `Share`
+  - `SortReply`
+  - `ApproveReply`
+  - `Collect`
+  - Coauthor stuff?
+  - Mentions?
+  - Private mentions?
+  - Media stuff?
+- Message
+  - `Read`
+  - `Send`
+  - `SeeContacts`
+  - `SeePresence`
+
+Account
+  - `ViewInfo`
+  - `ReadNotif`
+  - `ClearNotif`
+  - `ManageInfo`
+  - `ManageCredentials`
+  - `Grant`
+  - `Guest` 
+  - `Owner`
+
+## Structure
+
+Claims have to be represented somehow, and we're overwhelmingly using JWTs for that. As a first pass, perhaps this kind of claims structure:
+
+```json
+{
+  "kid": "blah",
+  "alg": "blah"
+}.{
+  "iss": "https://letterbook.example",
+  "exp": 1704088800,
+  "profiles": {
+    "coffee_nebula": ["Owner"],
+    "seven9": ["ApproveReply", "SortReply", "ApproveFollower", "Block", "Guest"]
```

Comment:
Thoughts on nested claims like in the doc, vs namespaced ones like 
```json
"profile:coffee_nebula": [],
"profile:seven9": [],
"account:katherine_janeway": []
```

Reply:
also, to be clear, these would all be UUIDs (or id25 representations of UUIDs). I used words here for readability

Reply:
Yeah. It is unbounded, but as a practical matter, I think it would be rare to have claims on *any* profile but your own, and very rare to have more than a handful of them. For accounts, I expect it would be exceedingly rare to have claims on any but your own. In fact, the only reason I wanted to support claims on multiple accounts in a token is to facilitate admins and moderators being able to look up contact info.

Perhaps supported by a wildcard entry? Like `"profile:*": ["Read"], "account:*": ["ViewInfo"]`? And that could possibly be done via an explicit and sort lived privilege escalation.